src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -246,6 +255,22 @@ Copyright (c) .NET Foundation. All rights reserved.
               TreatWarningsAsErrors="$(TreatWarningsAsErrors)"/>
   </Target>
 
+  <Target Name="_GenerateNuspecUsingProjectFile"
+          DependsOnTargets="_CalculateInputsOutputsForPack;_GetProjectReferenceVersions"
+          Condition="'$(IsPackable)' == 'true'"
+          Inputs="@(NuGetPackInput)" Outputs="@(NuGetPackOutput)">
+    <CreateProperty Value="true">
```

Comment:
i am not sure i understand why you are setting this property?

Reply:
I didn't want to duplicate the contents of the `_GenerateNuspecCommon` target (a 49 line call to `PackTask`), one for the conditions and Input/Output freshness on `_GenerateNuspecUsingProjectFile` and one for the conditions on `_GenerateNuspecUsingNuspecFile`. Creating a property in those two targets makes that property set only if the `_GenerateNuspecUsingProjectFile` target was selected and its conditions were satisfied, OR if the `_GenerateNuspecUsingNuspecFile` target was selected and *its* conditions were satisfied.

Reply:
I see, makes sense. Thanks for explaining. Does _GenerateNuspecCommon still need to depend on _CalculateInputsOutputsForPack & _GetProjectReferenceVersions ? I would imagine it shouldn't since those would have already run.